workers/default/ojs/tasks.py

```diff
@@ -20,32 +16,38 @@ class PublishToOJSTask(BaseTask):
 
     name = "publish_to_ojs"
 
-    def execute_task(self):
-        """Execute task intructions."""
+    def process_object(self, obj):
         work_path = self.get_work_path()
-        data = self.get_param('ojs_metadata')
+        ojs_metadata = self.get_param(self.get_param('params'))
 
         _, result = publish(os.path.join(work_path, 'ojs_import.xml'),
-                            data['ojs_journal_code'])
+                            ojs_metadata['ojs_journal_code'])
 
-        object_id = _generate_object_id(data['ojs_journal_code'], result)
-        return {'object_id': object_id}
+        children = obj.get_parts()
+        for i, child in enumerate(children):
+            child.metadata.ojs_id = \
```

Comment:
Statt `_generate_object_id()` zu löschen hätte ich der Funktion einen zusätzlichen Parameter `prefix` gegeben und sie einmal in der Schleife und einmal für das issue aufgerufen.

Reply:
Ich finds irgendwie Quatsch für einen simplen 1-Zeiler eine extra Funktion zu haben.

Reply:
Es sind ja jetzt schon zwei zeilen, die jetzt auch schon zweimal genutzt werden. Zusammen mit der Tatsache, dass genau die Funktion vorher schon da war das für mich Grund genug für einen Änderungswunsch ...